### PR TITLE
Baremetal platform tests refinement

### PR DESCRIPTION
**Pull request by andfasano**
This PR adds some more detailed expectations to the existing baremetal platform tests. It also provides some additional utility methods to make easier writing expectations on unstructured fields.

### CODE REVIEW

File: test/extended/baremetal/hosts.go [hunk 1]
```diff
@@ -33,6 +34,29 @@ func baremetalClient(dc dynamic.Interface) dynamic.ResourceInterface {
...
 }
 
+type FieldGetterFunc func(obj map[string]interface{}, fields ...string) (interface{}, bool, error)
+
+func expectField(host unstructured.Unstructured, nestedField string, fieldGetter FieldGetterFunc) o.Assertion {
+	fields := strings.Split(nestedField, ".")
+
+	value, found, err := fieldGetter(host.Object, fields...)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(found).To(o.BeTrue(), "baremetalhost field not found")
```
Comment (ardaguclu) [inline test/extended/baremetal/hosts.go:44]: instead "baremetalhost", nestedField can be used in here.

File: test/extended/baremetal/hosts.go [hunk 2]
```diff
@@ -33,6 +34,29 @@ func baremetalClient(dc dynamic.Interface) dynamic.ResourceInterface {
...
+	return o.Expect(value)
+}
+
+func expectStringField(host unstructured.Unstructured, nestedField string) o.Assertion {
```
Comment (ardaguclu) [inline test/extended/baremetal/hosts.go:48]: These wrappers simplify the expects and seems useful. But I wonder why I can not find any usage in the repo. Does nobody check nested fields in resources? or use different method?

File: test/extended/baremetal/hosts.go (hunk 1 above)
Comment (andfasano) [reply test/extended/baremetal/hosts.go:44]: Good point, thanks

File: test/extended/baremetal/hosts.go (hunk 2 above)
Comment (andfasano) [reply test/extended/baremetal/hosts.go:48]: There's another allowed approach for types that cannot be imported, and it's to copy and paste the struct and decode it. In this case I found easier to adopt the other recommended approach, ie to use `unstructured` pkg

**Comment by stbenjam** [comment]
Great to see more baremetal-specific checks

/lgtm

**Pull request merged**